creator/files/mutations/version.py

```diff
@@ -219,8 +223,132 @@ def mutate(
         return VersionUploadMutation(success=True, version=version)
 
 
+class ValidateFileVersionInput(graphene.InputObjectType):
+    """Parameters used when validating a file version against templates"""
+
+    file_version = graphene.ID(
+        required=True, description="The file version being validated"
+    )
+    template_version = graphene.ID(
+        description="The version of the data template to validate the "
+        "file against. Either the template_version or study must be "
+        "provided to validate the file version."
+    )
+    study = graphene.ID(
+        description="The study which the file version belongs to. Will be "
+        "used to find the study templates that match the file version. "
+        "Either the template_version or study must be provided to "
+        "validate the file version."
+    )
+
+
+class ValidateFileVersionMutation(graphene.Mutation):
```

Comment:
I might recommend splitting this into two mutations, maybe `ValidateTemplateMatch` and `SuggestTemplates`.  This way, the one mutation isn't overloaded into multiple forks that split on the provided parameters (not immediately evident to the client what's happening) and the client doesn't need to determine what happened based on the shape of the results (this is acting against the typing that graphql provides rather than with it).

If you have a `ValidateTemplateMatch`, you can very clearly tell the client to expect a True/False value for success.
If you have a `SuggestTemplates`, you can very clearly tell the client to expect a list of templates, perhaps ordered or with additional information.

Combining the two means you have to fork on the input parameters and the client will have to fork on the result whereas it would be better if both you and the client could just do one thing didn't have to make logic forks.